packages/javascript/bh-shared-ui/src/components/Disable2FADialog/Disable2FADialog.tsx

```diff
@@ -15,19 +15,28 @@
 // SPDX-License-Identifier: Apache-2.0
 
 import { Button, Dialog, DialogActions, DialogContent, DialogContentText, DialogTitle, TextField } from '@mui/material';
-import React, { useState } from 'react';
+import React from 'react';
 
 const Disable2FADialog: React.FC<{
     open: boolean;
     onClose: () => void;
     onCancel: () => void;
     onSave: (secret: string) => void;
     error?: string;
-}> = ({ open, onClose, onCancel, onSave, error }) => {
-    const [secret, setSecret] = useState('');
+    secret: string;
+    setSecret: (secret: string) => void;
```

Comment:
Recommend naming this prop `onSecretChange` or similar

Reply:
You use it in the event handler for the secret input itself which is why I think `onSecretChange` is appropriate.

If you want to clear the value of the secret when the dialog closes or the form is submitted you can do that by setting that value in the onClose, onCancel, onSave callbacks that get passed in as props.

Reply:
To that end I think the handleOnClose and handleOnCancel functions below are not really necessary

Reply:
Ahh, I see what you mean now. Line 59 of that file actually existed prior to my changes. I'm not sure why it was set up this way as I'm still a noob with React. I do think that the `setSecret` intent has become conflated due to my changes, though.